### ISSUE TITLE
ProjectSettings::globalize_path doesn't return full system path

### ISSUE THREAD

**Issue by Geequlim** [godotengine/godot#11145]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Godot 3.0 with latest commit on Arch Linux

**Issue description:**
* The `ProjectSettings::globalize_path` always just replace `res:/` to `.` we cann't get full project path any more.
* The right mouse menu option `Show In File Manager` in filesystem dock is totally broken

**Comment by Geequlim** [comment]
The resource path of the project is `.` returned here.
https://github.com/godotengine/godot/blob/master/core/project_settings.cpp#L328

ping @supagu  as git blame with f4994e750df6c8c19035249f7fdf89c02a109bf4

**Comment by supagu** [comment]
I don't think this fix is correct. I've put up what I think is a better commit: 
https://github.com/supagu/godot/commit/94c65d18369615c94cfb7c0a043880ae53b1e8c8

The problem is this @Geequlim's change assumes prev_dir == current_dir which may not be the case.
To get symlinks working I've added support for relative path changes without resolving the actual physical directory name, which allows current_dir to never know if it is in a symlink directory or not: hence I am using this line: String next_dir = current_dir + "/" + p_dir;
So the cause of this problem is that in the constructor the initial path is relative, so I have changed this to start with an absolute path, so we should never end up with a directory name starting with "."

I did a test and this fixes the Show in File Manager button.
